### PR TITLE
Detect IP conflicts before gateway detection to fix communication fail

### PR DESCRIPTION
**Pull request by cyclinder**
Detect IP conflicts before gateway detection to  fix gateway detection could potentially update the arp table causing communication fail

# Thanks for contributing!

**Notice**:

* [ ] unite test or E2E test
* [ ] do not forget essential code comment and log
* [ ] document for the PR
* [ ] release note label
  "release/none"
  "release/bug"
  "release/feature"
* [ ] read about  Contribution notice: <https://spidernet-io.github.io/spiderpool/latest/develop/contributing/>

**What issue(s) does this PR fix**:

Fixes https://github.com/spidernet-io/spiderpool/issues/4475

**Special notes for your reviewer**:

### CODE REVIEW

File: cmd/coordinator/cmd/command_add.go [hunk 1]
```diff
@@ -189,8 +189,56 @@ func CmdAdd(args *skel.CmdArgs) (err error) {
 
 	logger.Sugar().Infof("Get coordinator config: %+v", c)
 
-	errg := errgroup.Group{}
-	//  we do detect gateway connection firstly
+	errgConflict := errgroup.Group{}
```
Comment (weizhoublue) [inline cmd/coordinator/cmd/command_add.go:192]: 补充 code comment，关于 每一步先后循序 原因说明 
arp 问题，引用 issue来详细说明

Comment (cyclinder) [reply cmd/coordinator/cmd/command_add.go:192]: done

File: cmd/coordinator/cmd/command_add.go [hunk 2]
```diff
@@ -189,8 +189,58 @@ func CmdAdd(args *skel.CmdArgs) (err error) {
...
+	// communication problems when IP conflict detection fails
+	// see https://github.com/spidernet-io/spiderpool/issues/4475
+	var ipc *ipchecking.IPChecker
+	if conf.IPConflict != nil && *conf.IPConflict {
+		logger.Debug("Try to detect ip conflict")
+		ipc, err = ipchecking.NewIPChecker(conf.DetectOptions.Retry, conf.DetectOptions.Interval, conf.DetectOptions.TimeOut, c.hostNs, c.netns, logger)
+		if err != nil {
+			return fmt.Errorf("failed to run NewIPChecker: %w", err)
+		}
+		ipc.DoIPConflictChecking(prevResult.IPs, c.currentInterface, &errgConflict)
```
Comment (weizhoublue) [inline cmd/coordinator/cmd/command_add.go:204]: 这个超时控制 在哪体现的

File: cmd/coordinator/cmd/command_add.go [hunk 3]
```diff
@@ -226,21 +275,10 @@ func CmdAdd(args *skel.CmdArgs) (err error) {
...
-		}
-		ipc.DoIPConflictChecking(prevResult.IPs, c.currentInterface, &errg)
-	} else {
-		logger.Debug("disable detect ip conflict")
-	}
-
-	if err = errg.Wait(); err != nil {
-		logger.Error("failed to detect gateway and ip checking", zap.Error(err))
-		if errors.Is(err, constant.ErrIPConflict) || errors.Is(err, constant.ErrGatewayUnreachable) {
+	if err = errgGateway.Wait(); err != nil {
```
Comment (weizhoublue) [inline cmd/coordinator/cmd/command_add.go:278]: 同理 检测 网关的 超时控制，在哪

File: cmd/coordinator/cmd/command_add.go (hunk 2 above)
Comment (cyclinder) [reply cmd/coordinator/cmd/command_add.go:204]: ipc, err = ipchecking.NewIPChecker(conf.DetectOptions.Retry, conf.DetectOptions.Interval, conf.DetectOptions.TimeOut, c.hostNs, c.netns, logger)

Comment (weizhoublue) [reply cmd/coordinator/cmd/command_add.go:204]: 这默认超时时间多少，在代码上 comment 下，这个变量 不容易 查源头的值

Comment (cyclinder) [reply cmd/coordinator/cmd/command_add.go:204]: 这些信息会在 final configuration 中打印，类似：

```
{"level":"DEBUG","ts":"2024-12-31T09:41:43.783Z","logger":"coordinator","caller":"cmd/command_add.go:84","msg":"final configuration","Action":"ADD","ContainerID":"17dc097c71526a6ab4ac53c07e48cc43e30d9b13bcac603ac2891eb23921dd2a","Netns":"/var/run/netns/cni-************************************","IfName":"net1","PodName":"dep-name-4ef71f347c-5496b8784b-r9s2v","PodNamespace":"ns-7dff59fa8a","conf":{"cniVersion":"0.3.1","name":"test-multus-6b86b94379","type":"coordinator","ipam":{},"dns":{},"detectGateway":false,"podDefaultCniNic":"eth0","overlayPodCIDR":["***********/18","fd00:10:243::/112"],"serviceCIDR":["**********/18","fd00:10:233::/116"],"hijackCIDR":["***********/16"],"tunePodRoutes":true,"mode":"overlay","hostRuleTable":500,"podRPFilter":0,"txQueueLen":0,"detectIPConflict":true,"detectOptions":{"retries":3,"interval":"1s","timeout":"3s"},"logOptions":{"logLevel":"debug","logFile":"/var/log/spidernet/coordinator.log","logMaxSize":0,"logMaxAge":0,"logMaxCount":0}}}
```

> 缺省值 是 1s ？太长了

多少合适？100ms？

Comment (weizhoublue) [reply cmd/coordinator/cmd/command_add.go:204]: 100ms 足够够了，并且最好要 实测下

**Review by weizhoublue — APPROVED** [review]

**Pull request merged**